sklearn/model_selection/tests/test_validation.py

```diff
@@ -936,6 +937,124 @@ def test_cross_val_predict_with_method():
         assert_array_almost_equal(expected_predictions, predictions)
 
 
+def test_cross_val_predict_corner_case():
```

Comment:
Seems better. Also, I just thought of something. Since this is a test for a rare edge case, how about `test_cross_val_predict_edge_case`?